uber/custom_tags.py

```diff
@@ -500,42 +492,34 @@ def render(self, context):
         return prices
 
 
-class Notice(template.Node):
-    def notice(self, label, takedown, discount=0, amount_extra=0):
-        for day, price in sorted(c.PRICE_BUMPS.items()):
-            if day < takedown and localized_now() < day:
-                return 'Price goes up to ${} at 11:59pm EST on {}'.format(price + amount_extra - discount, (day - timedelta(days=1)).strftime('%A, %b %e'))
-            elif localized_now() < day:
-                return '{} closes at 11:59pm EST on {}. Price goes up to ${} at-door.'.format(label, takedown.strftime('%A, %b %e'), price + amount_extra - discount, (day - timedelta(days=1)).strftime('%A, %b %e'))
-        return '{} closes at 11:59pm EST on {}'.format(label, takedown.strftime('%A, %b %e'))
-
-
-@tag
-class attendee_price_notice(Notice):
-    def render(self, context):
-        return self.notice('Preregistration', c.PREREG_TAKEDOWN)
-
-
-@tag
-class group_price_notice(Notice):
-    def render(self, context):
-        return self.notice('Group preregistration', c.GROUP_PREREG_TAKEDOWN, discount=c.BADGE_PRICES['group_discount'])
+@register.tag(name='price_notice')
+def price_notice(parser, token):
+    return PriceNotice(*token.split_contents()[1:])
 
 
-@tag
-class supporter_price_notice(Notice):
-    def render(self, context):
-        return self.notice('Supporter preregistration', c.SUPPORTER_DEADLINE, amount_extra=c.SUPPORTER_LEVEL)
```

Comment:
This is just an adaptation of what we had before.  Instead of having separate custom tags for the different notices, all of the parameters are just specified in the template now.  This is more flexible, since it means that if a plugin adds a custom level they don't need to implement a new tag.